Scripts/main.js

```diff
@@ -0,0 +1,181 @@
+'use strict'
+
+var Rule = function (value) {
+    this.value = value;
+    this.functionArray = [];
+}
+
+Rule.prototype.setValue = function (value) {
+    this.value = value;
+}
+
+Rule.prototype.isRequired = function () {
+    this.functionArray.push(function (ruleValue) {
+        var isValid = ruleValue ? true : false;
+        var message = "Value must be different from " + ruleValue;
+        var validationResult = {
+            errorMessage: isValid ? null : message,
+            isValid: isValid
+        }
+        return validationResult;
+    });
+    return this;
+}
+
+Rule.prototype.min = function (value) {
+    var parametr = value;
+    this.functionArray.push(function (ruleValue, ruleName) {
```

Comment:
Is ruleName required here?
You are inside the rule, so I think it's redundant to pass rule name from the outside, because you know the rule name

Reply:
It's unclear variable name. It name of the property to be ruled.